src/QiParsers.h

```diff
@@ -100,14 +101,14 @@ inline bool parseNumber(char decimalMark, char groupingMark, Iterator& first,
       break;
     case STATE_FIN:
       last = cur++;
-      res = sum;
+      res = negative ? sum * -1 : sum;
```

Comment:
It could, I guess the only issue is it wouldn't break out of the for loop, so you would iterate through the rest of the string needlessly (probably doesn't matter).

It could be a goto though...